src/programs/constrained_search/utils/get_centre_vector.py

```diff
@@ -0,0 +1,228 @@
+"""Calculates the centre vector between two PDB structures."""
+
+import argparse
+import math
+
+import mmdf
+import numpy as np
+import pandas as pd
+import roma
+import torch
+
+
+def calculate_mean_position(df: pd.DataFrame) -> torch.Tensor:
+    """
+    Calculate the mean position of a PDB structure.
+
+    Attributes
+    ----------
+    df: pd.DataFrame
+        The mmdf DataFrame with the PDB structure.
+
+    Returns
+    -------
+    torch.Tensor
+        The mean position as a 3D vector [x, y, z].
+    """
+    # Extract coordinates and calculate mean
+    coords = torch.tensor(df[["x", "y", "z"]].values, dtype=torch.float32)
+    mean_pos = coords.mean(dim=0)
+
+    return mean_pos
+
+
+def rotate_vector(vector: torch.Tensor, rotation_matrix: torch.Tensor) -> torch.Tensor:
+    """
+    Rotate a vector using a rotation matrix.
+
+    Attributes
+    ----------
+    vector: torch.Tensor
+        The vector to rotate.
+    rotation_matrix: torch.Tensor
+        The rotation matrix to use.
+
+    Returns
+    -------
+        Rotated vector
+    """
+    # Convert to tensor if not already
+    if not isinstance(vector, torch.Tensor):
+        vector = torch.tensor(vector, dtype=torch.float32)
+
+    # Apply rotation
+    rotated_vector = rotation_matrix @ vector
+
+    return rotated_vector
+
+
+def vector_to_euler(
+    vector: torch.Tensor,
+) -> tuple[float, float, float]:
+    """
+    Convert a vector to Euler angles.
+
+    Attributes
+    ----------
+    vector: torch.Tensor
+        The vector to convert to Euler angles.
+
+    Returns
+    -------
+    tuple[float, float, float]
+        The Euler angles in degrees (phi, theta, psi).
+    """
+    # Normalize the vector
+    norm_vector = vector / torch.norm(vector)
+
+    # Calculate theta as the angle from the z-axis
+    theta = torch.acos(norm_vector[2])
+
+    # Calculate phi as the azimuthal angle in the x-y plane
+    phi = torch.atan2(norm_vector[1], norm_vector[0])
+
+    # For a direction vector, psi can be set to 0
+    psi = torch.tensor(0.0)
+
+    # Convert to degrees
+    phi_deg = phi.item() * 180 / math.pi
```

Comment:
Actually, instead of using the `math.pi` constant or `np.pi` constant, use the built-in `np.rad2deg` function for readability.